nym-wallet/src/context/bonding.tsx

```diff
@@ -185,82 +186,145 @@ export const BondingContextProvider: FCWithChildren = ({ children }): JSX.Elemen
     setBondedNode(undefined);
   };
 
-  const getAdditionalMixnodeDetails = async (mixId: number) => {
-    const additionalDetails: {
+  /**
+   * Fetch mixnode **optional** data.
+   * ⚠ The underlying queries are allowed to fail.
+   */
+  const fetchMixnodeDetails = async (mixId: number, host: string, port: number) => {
+    const details: {
       status: MixnodeStatus;
       stakeSaturation: string;
       estimatedRewards?: DecCoin;
       uptime: number;
+      averageUptime?: number;
+      setProbability?: InclusionProbabilityResponse;
+      nodeDescription?: TNodeDescription | undefined;
+      operatorRewards?: DecCoin;
       uncappedSaturation?: number;
     } = {
       status: 'not_found',
       stakeSaturation: '0',
       uptime: 0,
     };
 
-    try {
-      const statusResponse = await getMixnodeStatus(mixId);
-      const uptime = await getMixnodeUptime(mixId);
-      additionalDetails.status = statusResponse.status;
-      additionalDetails.uptime = uptime;
-    } catch (e) {
-      Console.log('getMixnodeStatus fails', e);
+    const requests = {
+      getMixnodeStatus: () => getMixnodeStatus(mixId),
+      getMixnodeUptime: () => getMixnodeUptime(mixId),
+      getMixnodeStakeSaturation: () => getMixnodeStakeSaturation(mixId),
+      getMixnodeRewardEstimation: () => getMixnodeRewardEstimation(mixId),
+      getInclusionProbability: () => getInclusionProbability(mixId),
+      getMixnodeAvgUptime: () => getMixnodeAvgUptime(),
+      getNodeDescription: () => getNodeDescriptionRequest(host, port),
+      getPendingOperatorRewards: () => getPendingOperatorRewards(clientDetails?.client_address || ''),
+    };
+
+    const promises = await Promise.allSettled([
+      requests.getMixnodeStatus(),
+      requests.getMixnodeUptime(),
+      requests.getMixnodeStakeSaturation(),
+      requests.getMixnodeRewardEstimation(),
+      requests.getInclusionProbability(),
+      requests.getMixnodeAvgUptime(),
+      requests.getNodeDescription(),
+      requests.getPendingOperatorRewards(),
+    ]);
+
+    promises.forEach((res, index) => {
+      const reqNames = Object.keys(requests);
+      if (res.status === 'rejected') {
+        Console.warn(`${reqNames[index]} request fails`, res.reason);
+      }
+    });
```

Comment:
```suggestion
    promises.forEach((res, index) => {
      if (res.status === 'rejected') {
        Console.warn('request fails', res.reason);
      }
    });
```

Reply:
I think the failure reason should give us some details about the request that failed?